### ISSUE TITLE
Modify Vx suite to handle full RRFS NA 3km domain

### ISSUE THREAD

**Issue by JeffHamiltonNOAA** [NOAA-GSL/VxLegacyIngest#508]
With the upcoming RRFS_A switch to the full NA 3km domain, it is now time to put a full effort into modifying our scripts to handle the volume of data associated with it. Upper air and aircraft are the two main focus areas here with the 3-D component complicating the issue

**Comment by JeffHamiltonNOAA** [comment]
I have an Upper Air executable that processes the full 3km NA domain. I'll begin making the changes elsewhere to allow for retrospective and real time execution across all verification types

**Comment by bonnystrong** [comment]
Ming is waiting on receiving grid data from Ming.

**Comment by bonnystrong** [comment]
Modelers are indicating that they are ok to wait until Jeff's return.  Molly working on scorecard.

**Comment by JeffHamiltonNOAA** [comment]
After speaking with Isidora and Daniel Abdi, we have determined that the interpolation code Daniel has come up with may be a better solution than our legacy code as it now stands. I am working to install the necessary libraries (C++ eigen and eccodes 2.28.0) to build his executable, available in the following repository:

https://github.com/danielabdi-noaa/interp

The capability to interpolate to specific station locations and output a grib file with that subset of data is very intriguing and should resolve all of our issues if it works as advertised

**Comment by JeffHamiltonNOAA** [comment]
I've installed updated versions of eccodes, ecbuild, and eigen on Jet in the role.amb-verif home directory. I'm still working on the final install of Daniel's interp executable, but CMake is haivng trouble finding the location of eccodes even though I'm passing it to the build command. I've reached out to Daniel for assistance

**Comment by JeffHamiltonNOAA** [comment]
Hallelujah! I've managed to compile Daniel's code on Jet and will now test using our scripts as a baseline

**Comment by JeffHamiltonNOAA** [comment]
I've tested Daniels code on the front end nodes and am not optimistic with the results. Grabbing 10 variables (we will eventually need up to 60 for full column UA verification) takes around 10 minutes per file. Utilizing MPI doesn't work well on the front end nodes with the I/O issues (The same 10 variables took almost an hour to process with 3 threads) and I would need to completely redesign the ingest in order to take advantage of the Big Memory nodes on Jet to POSSIBLY make this faster... It may just be worth trying to make the old binary output methodology work instead... Frustrating

**Comment by bonnystrong** [comment]
Going through all our projection code because that must be where the problem is; all else has been checked.

**Comment by JeffHamiltonNOAA** [comment]
I've modified all scripts for the 3km domain (grid 23 in our scripts), but am now testing the new projection routine against the 13km domain as I believe that the slightly larger than expected errors for RRFS_NA_13km could be an artifact of using the old projection method (for RAP). Once done, this issue will be complete

**Comment by JeffHamiltonNOAA** [comment]
I've determined that the 13km domain will need this new projection transformation function as well. I've modified all scripts as best I can in anticipation of a NA 3km domain, but will likely need to test properly once real time data is flowing. That will be a new issue, this one can be closed
